domenetjenester/registerinnhenting/src/main/java/no/nav/foreldrepenger/domene/registerinnhenting/MorsAktivitetInnhenter.java

```diff
@@ -78,111 +93,147 @@ public void innhentMorsAktivitet(Behandling behandling) {
+    private static LocalDate minsteFraDatoMinusIntervall(List<OppgittPeriodeEntitet> aktuellePerioder) {
         return aktuellePerioder.stream()
             .map(OppgittPeriodeEntitet::getFom)
             .min(LocalDate::compareTo)
-            .map(minDato -> minDato.minusWeeks(2));
+            .map(minDato -> minDato.minusWeeks(2))
+            .orElseThrow();
     }
 
-    private static List<OppgittPeriodeEntitet> hentPerioderMedAktivitetskrav(YtelseFordelingAggregat ytelseaggregat, FagsakYtelseType ytelseType) {
+    private static List<OppgittPeriodeEntitet> hentPerioderMedAktivitetskrav(YtelseFordelingAggregat ytelseaggregat) {
         return ytelseaggregat.getGjeldendeFordeling()
             .getPerioder()
             .stream()
-            .filter(p -> FagsakYtelseType.FORELDREPENGER.equals(ytelseType) && UttakPeriodeType.FELLESPERIODE.equals(p.getPeriodeType()) && MorsAktivitet.ARBEID.equals(p.getMorsAktivitet()))
+            .filter(p -> skalInnhenteAktivitetskravGrunnlag(p, ytelseaggregat))
             .toList();
     }
 
+    private static boolean skalInnhenteAktivitetskravGrunnlag(OppgittPeriodeEntitet p, YtelseFordelingAggregat ytelseaggregat) {
+        if (!MorsAktivitet.ARBEID.equals(p.getMorsAktivitet())) {
+            return false;
+        }
+        return !ytelseaggregat.harAnnenForelderRett() ? Objects.equals(UttakPeriodeType.FORELDREPENGER, p.getPeriodeType()) : Objects.equals(
+            FELLESPERIODE, p.getPeriodeType());
+    }
+
     private AktivitetskravArbeidPerioderEntitet lagAktivitetskravPerioderBuilder(Map<String, List<ArbeidsforholdMedPermisjon>> mapAvOrgnrOgAvtaler,
-                                                                                         LocalDate fraDato,
-                                                                                         LocalDate tilDato) {
+                                                                                 LocalDate fraDato,
+                                                                                 LocalDate tilDato) {
         List<AktivitetskravArbeidPeriodeEntitet.Builder> builderListe = new ArrayList<>();
-        mapAvOrgnrOgAvtaler.forEach((key, value) -> {
+        mapAvOrgnrOgAvtaler.forEach((orgnr, value) -> {
             var stillingsprosentTidslinje = stillingsprosentTidslinje(value);
             var permisjonProsentTidslinje = permisjonTidslinje(value);
             var grunnlagTidslinje = stillingsprosentTidslinje.crossJoin(permisjonProsentTidslinje,
                     bigDecimalTilAktivitetskravVurderingGrunnlagCombinator())
                 .crossJoin(lagTidslinjeMed0(fraDato, tilDato), StandardCombinators::coalesceLeftHandSide)
                 .compress();
 
-            builderListe.addAll(grunnlagTidslinje.stream().map(segment -> opprettBuilder(segment, key)).toList());
+            builderListe.addAll(grunnlagTidslinje.stream().map(segment -> opprettBuilder(segment, orgnr)).toList());
```

Comment:
1. tror vi kan erstatte .crossJoin(lagTidslinjeMed0(fraDato, tilDato) med en .intersection(new LDI(fra, til))
2. pga BigDecimal/equals trengs en kustom compressor .compress(LocalDateInterval::abutsWorkdays, <se nedenfor>, StandardCombinators::leftOnly)
3. Note to self: Lage en standard SegmentSplitter<OppgittPeriodeEntitet> som kan brukes i new LDTimeline

Nedenfor: (v1, v2) -> v1.stilling.compareTo(v2.stilling) == 0 && v1.permp.compareTo(v2.perm2) == 0 && v1.permt.equals(v2.permt)

Reply:
1. Tror ikke vi kan endre fra crossjoin til intersection her. Vi vil ha nullobject i tidsintervaller som ikke finnes i tidslinja
2. fixed